xcodeproj/internal/xcode_targets.bzl

```diff
@@ -261,44 +270,81 @@ def _to_xcode_target_product(product):
         _additional_files = product.framework_files,
     )
 
-def _merge_xcode_target(*, src, dest):
+def _merge_xcode_target(*, src_swift, src_other, dest):
     """Creates a new `xcode_target` by merging the values of `src` into `dest`.
 
     Args:
-        src: The `xcode_target` to merge into `dest`.
+        src_swift: The `xcode_target` to merge into `dest` that `is_swift`.
+        src_other: The `xcode_target` to merge into `dest` that is not `is_swift`.
         dest: The `xcode_target` being merged into.
 
     Returns:
         A value as returned by `xcode_targets.make`.
     """
 
-    build_settings = dict(structs.to_dict(src._build_settings))
+    build_settings = {}
+
+    if src_swift:
+        build_settings = dicts.add(
+            structs.to_dict(src_swift._build_settings),
+            build_settings,
+        )
+    if src_other:
+        build_settings = dicts.add(
+            structs.to_dict(src_other._build_settings),
+            build_settings,
+        )
+
+    # This is intentionally last since it's the highest-level target that should
+    # override some build settings from the other target(s).
     build_settings = dicts.add(
         structs.to_dict(dest._build_settings),
         build_settings,
     )
 
+    srcs = []
+    platform = None
+    conlyopts = dest._conlyopts
+    cxxopts = dest._cxxopts
+    swiftcopts = dest._swiftcopts
+    swiftmodules = []
+    modulemaps = ()
+    if src_swift:
+        srcs.append(src_swift)
+        swiftcopts = src_swift._swiftcopts or dest._swiftcopts
+        platform = src_swift.platform
+        swiftmodules = src_swift._swiftmodules
+        modulemaps = src_swift._modulemaps
+    if src_other:
+        srcs.append(src_other)
+        conlyopts = src_other._conlyopts or dest._conlyopts
+        cxxopts = src_other._cxxopts or dest._cxxopts
+        platform = src_other.platform
+        modulemaps = modulemaps + (src_other._modulemaps if modulemaps else src_other._modulemaps)
```

Comment:
I thought `modulemaps` only comes from and is only used by the swift code?

Reply:
This fixed some issue in different examples/fixtures. I can try to figure out what that was/if it's necessary again though.